File: fetch_images.py
import urllib.request
import re
import os
import ssl

url = 'https://www.cruzcoaching.net/our-team'
req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
ctx = ssl.create_default_context()
ctx.check_hostname = False
ctx.verify_mode = ssl.CERT_NONE

try:
    html = urllib.request.urlopen(req, context=ctx).read().decode('utf-8')
    img_matches = re.findall(r'<img[^>]+src=(["\'])(.*?)\1', html)
    data_matches = re.findall(r'<img[^>]+data-image=(["\'])(.*?)\1', html)
    
    urls = set(m[1] for m in img_matches + data_matches)
    
    os.makedirs('downloaded_images', exist_ok=True)
    
    i = 0
    for src in urls:
        if src.startswith('//'): src = 'https:' + src
        elif src.startswith('/'): src = 'https://www.cruzcoaching.net' + src
        
        print(f"Downloading: {src}")
        try:
            req_img = urllib.request.Request(src, headers={'User-Agent': 'Mozilla/5.0'})
            img_data = urllib.request.urlopen(req_img, context=ctx).read()
            with open(f"downloaded_images/img_{i}.jpg", "wb") as f:
                f.write(img_data)
            i += 1
        except Exception as e:
            print(f"Failed {src}: {e}")
except Exception as e:
    print(f"Error: {e}")

File: fetch_images.mjs
import fs from 'fs';
import path from 'path';

const url = 'https://www.cruzcoaching.net/our-team';

async function run() {
  const res = await fetch(url);
  const text = await res.text();
  
  const imgRegex = /<img[^>]+src=(["'])([^"']+)\1/g;
  const dataImageRegex = /<img[^>]+data-image=(["'])([^"']+)\1/g;
  
  let match;
  const imageUrls = new Set();
  
  while ((match = imgRegex.exec(text)) !== null) {
      imageUrls.add(match[2]);
  }
  while ((match = dataImageRegex.exec(text)) !== null) {
      imageUrls.add(match[2]);
  }
  
  if (!fs.existsSync('downloaded_images')) {
      fs.mkdirSync('downloaded_images');
  }
  
  let i = 0;
  for (let src of imageUrls) {
      if (src.startsWith('//')) src = 'https:' + src;
      else if (src.startsWith('/')) src = 'https://www.cruzcoaching.net' + src;
      
      try {
          console.log(`Downloading: ${src}`);
          const imgRes = await fetch(src);
          const buffer = await imgRes.arrayBuffer();
          fs.writeFileSync(`downloaded_images/img_${i}.jpg`, Buffer.from(buffer));
          i++;
      } catch (e) {
          console.error(`Failed ${src}`, e.message);
      }
  }
}

run();
